cmd/server/main.go

```diff
@@ -397,6 +400,18 @@ func SetupSignalHandler() (stopCh <-chan struct{}) {
 	return stop
 }
 
+func getUDSListener(ctx context.Context, udsName string) (net.Listener, error) {
+	udsListenerLock.Lock()
+	defer udsListenerLock.Unlock()
+	defer syscall.Umask(syscall.Umask(0007))
```

Comment:
Seems a bit sneaky to rely on parameter evaluation of a deferred function to get the first Umask call to happen before the rest of the function. Going to lgtm but we should consider either comments or breaking it out to make it a little more obvious to anyone reading the code.

Reply:
Broke it out to make the code more obvious. Thanks!